src/scripting/lua_game.cc

```diff
@@ -667,21 +667,24 @@ int LuaPlayer::reveal_fields(lua_State* L) {
 }
 
 /* RST
-   .. method:: hide_fields(fields[, unexplore = false])
+   .. method:: hide_fields(fields[, mode = "seen"])
 
       Undo the effect of :meth:`reveal_fields` on these fields for the current player and
-      optionally completely hide them.
+      optionally completely hide them or reset them to unexplored.
       See also :ref:`field_animations` for animated hiding.
 
       :arg fields: The fields to hide
       :type fields: :class:`array` of :class:`fields <wl.map.Field>`
 
-      :arg unexplore: *Optional*. If  ``true``, the fields will be marked as completely unexplored
+      :arg state: *Optional*. If  ``permanent``, the fields will be marked as completely hidden
          and will not be seen by buildings or workers until they are revealed again
          by :meth:`reveal_fields`.
-         If `false`, They will no longer be permanently visible, but can still be seen by
-         buildings or workers (own or allied), and the player will remember the last seen state.
-      :type unexplore: :class:`boolean`
+         If ``seen``, They will no longer be permanently visible, but can still be seen by
+         buildings or workers (own or allied), and the player will remember the last state that
+         they had been seen. this is the default
+         If ``explorable``, They will no longer be visible, but can still be rediscovered by
+         buildings, ships or workers (own or allied).
+      :type state: :class:`string`
```

Comment:
This would require changing all occurences of hide fields then, which was not desired. The previously seen stste was the default until this change if no boolean was provided.

Reply:
Maybe i do not understand enough of this vision things.

"seen" sounds to me they will be half-visible, like here:

![half-visible](https://user-images.githubusercontent.com/4470900/111062126-5d9e6a80-84a7-11eb-8f11-6a8bcb7f1aa0.png)

If this i not true, which value is used to get something like that?

Reply:
Forget it, i got it now :-)